src/main/java/com/commercetools/sync/cartdiscounts/CartDiscountSync.java

```diff
@@ -202,12 +204,21 @@ private void handleError(@Nonnull final String errorMessage,
                         .resolveReferences(newCartDiscount)
                         .thenCompose(resolvedDraft -> syncDraft(oldCartDiscountMap, resolvedDraft))
                         .exceptionally(completionException -> {
-                            final ReferenceResolutionException referenceResolutionException =
-                                (ReferenceResolutionException) completionException.getCause();
-                            final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES, newCartDiscount.getKey(),
-                                             referenceResolutionException.getMessage());
-                            handleError(errorMessage, referenceResolutionException, 1);
-                            return null;
+                            if (completionException.getCause() instanceof ReferenceResolutionException) {
+                                final ReferenceResolutionException referenceResolutionException =
```

Comment:
please refactor the lambda expression in here, this is totally not easy to understand, my suggestion is moving those checks to another method with a good name.

Reply:
As discussed with Roman, the thrown exception will be passed into handleError directly and let user to deal with it. If-else checking is no longer used in Sync classes.